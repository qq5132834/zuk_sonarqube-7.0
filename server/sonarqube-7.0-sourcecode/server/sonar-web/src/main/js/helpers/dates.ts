/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import {
  differenceInDays as _differenceInDays,
  differenceInSeconds as _differenceInSeconds,
  differenceInYears as _differenceInYears,
  isSameDay as _isSameDay,
  parse,
  startOfDay as _startOfDay
} from 'date-fns';

function pad(number: number) {
  if (number < 10) {
    return '0' + number;
  }
  return number;
}

type ParsableDate = string | number | Date;

export function parseDate(rawDate: ParsableDate): Date {
  return parse(rawDate);
}

export function toShortNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return `${date.getFullYear()}-${pad(date.getMonth() + 1)}-${pad(date.getDate())}`;
}

export function toNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return date.toISOString().replace(/\..+Z$/, '+0000');
}

export function startOfDay(date: Date): Date {
  return _startOfDay(date);
}

export function isValidDate(date: Date): boolean {
  return !isNaN(date.getTime());
}

export function isSameDay(dateLeft: Date, dateRight: Date): boolean {
  return _isSameDay(dateLeft, dateRight);
}

export function differenceInYears(dateLeft: Date, dateRight: Date): number {
  return _differenceInYears(dateLeft, dateRight);
}

export function differenceInDays(dateLeft: Date, dateRight: Date): number {
  return _differenceInDays(dateLeft, dateRight);
}

export function differenceInSeconds(dateLeft: Date, dateRight: Date): number {
  return _differenceInSeconds(dateLeft, dateRight);
}
