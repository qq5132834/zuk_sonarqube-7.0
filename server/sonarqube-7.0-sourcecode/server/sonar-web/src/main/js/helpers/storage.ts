/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
const PROJECTS_DEFAULT_FILTER = 'sonarqube.projects.default';
const PROJECTS_FAVORITE = 'favorite';
const PROJECTS_ALL = 'all';

const PROJECTS_VIEW = 'sonarqube.projects.view';
const PROJECTS_VISUALIZATION = 'sonarqube.projects.visualization';
const PROJECTS_SORT = 'sonarqube.projects.sort';

const PROJECT_ACTIVITY_GRAPH = 'sonarqube.project_activity.graph';
const PROJECT_ACTIVITY_GRAPH_CUSTOM = 'sonarqube.project_activity.graph.custom';

function save(key: string, value?: string): void {
  try {
    if (value) {
      window.localStorage.setItem(key, value);
    } else {
      window.localStorage.removeItem(key);
    }
  } catch (e) {
    // usually that means the storage is full
    // just do nothing in this case
  }
}

export function saveFavorite(): void {
  save(PROJECTS_DEFAULT_FILTER, PROJECTS_FAVORITE);
}

export function isFavoriteSet(): boolean {
  const setting = window.localStorage.getItem(PROJECTS_DEFAULT_FILTER);
  return setting === PROJECTS_FAVORITE;
}

export function saveAll(): void {
  save(PROJECTS_DEFAULT_FILTER, PROJECTS_ALL);
}

export function isAllSet(): boolean {
  const setting = window.localStorage.getItem(PROJECTS_DEFAULT_FILTER);
  return setting === PROJECTS_ALL;
}

export function saveView(view?: string): void {
  save(PROJECTS_VIEW, view);
}

export function getView(): string | null {
  return window.localStorage.getItem(PROJECTS_VIEW);
}

export function saveVisualization(visualization?: string): void {
  save(PROJECTS_VISUALIZATION, visualization);
}

export function getVisualization(): string | null {
  return window.localStorage.getItem(PROJECTS_VISUALIZATION);
}

export function saveSort(sort?: string): void {
  save(PROJECTS_SORT, sort);
}

export function getSort(): string | null {
  return window.localStorage.getItem(PROJECTS_SORT);
}

export function saveCustomGraph(metrics?: string[]): void {
  save(PROJECT_ACTIVITY_GRAPH_CUSTOM, metrics ? metrics.join(',') : '');
}

export function getCustomGraph(): string[] {
  const customGraphs = window.localStorage.getItem(PROJECT_ACTIVITY_GRAPH_CUSTOM);
  return customGraphs ? customGraphs.split(',') : [];
}

export function saveGraph(graph?: string): void {
  save(PROJECT_ACTIVITY_GRAPH, graph);
}

export function getGraph(): string {
  return window.localStorage.getItem(PROJECT_ACTIVITY_GRAPH) || 'issues';
}
