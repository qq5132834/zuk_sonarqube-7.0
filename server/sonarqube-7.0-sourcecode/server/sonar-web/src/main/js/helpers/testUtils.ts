/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow, ShallowRendererProps, ShallowWrapper, ReactWrapper } from 'enzyme';
import { IntlProvider } from 'react-intl';

export const mockEvent = {
  target: { blur() {} },
  currentTarget: { blur() {} },
  preventDefault() {},
  stopPropagation() {}
};

export function click(element: ShallowWrapper | ReactWrapper, event = {}): void {
  element.simulate('click', { ...mockEvent, ...event });
}

export function clickOutside(event = {}): void {
  const dispatchedEvent = new MouseEvent('click', event);
  window.dispatchEvent(dispatchedEvent);
}

export function submit(element: ShallowWrapper): void {
  element.simulate('submit', {
    preventDefault() {}
  });
}

export function change(element: ShallowWrapper, value: string, event = {}): void {
  element.simulate('change', {
    target: { value },
    currentTarget: { value },
    ...event
  });
}

export function keydown(keyCode: number): void {
  const event = new KeyboardEvent('keydown', { keyCode } as KeyboardEventInit);
  document.dispatchEvent(event);
}

export function elementKeydown(element: ShallowWrapper, keyCode: number): void {
  const event = {
    currentTarget: { element },
    keyCode,
    preventDefault() {}
  };

  if (typeof element.type() === 'string') {
    // `type()` is string for native dom elements
    element.simulate('keydown', event);
  } else {
    element.prop<Function>('onKeyDown')(event);
  }
}

export function doAsync(fn?: Function): Promise<void> {
  return new Promise(resolve => {
    setImmediate(() => {
      if (fn) {
        fn();
      }
      resolve();
    });
  });
}

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider({ locale: 'en' }, {});
const { intl } = intlProvider.getChildContext();
export function shallowWithIntl(node: React.ReactElement<any>, options: ShallowRendererProps = {}) {
  return shallow(node, { ...options, context: { intl, ...options.context } });
}
