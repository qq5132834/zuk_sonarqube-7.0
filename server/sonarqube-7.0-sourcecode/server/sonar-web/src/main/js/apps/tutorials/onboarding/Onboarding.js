/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import TokenStep from './TokenStep';
import OrganizationStep from './OrganizationStep';
import AnalysisStep from './AnalysisStep';
import ProjectWatcher from './ProjectWatcher';
import { skipOnboarding } from '../../../api/users';
import { translate, translateWithParameters } from '../../../helpers/l10n';
import { getProjectUrl } from '../../../helpers/urls';
import handleRequiredAuthentication from '../../../app/utils/handleRequiredAuthentication';
import './styles.css';

/*::
type Props = {|
  currentUser: { login: string, isLoggedIn: boolean },
  onFinish: () => void,
  organizationsEnabled: boolean,
  sonarCloud: boolean
|};
*/

/*::
type State = {
  finished: boolean,
  organization?: string,
  projectKey?: string,
  skipping: boolean,
  step: string,
  token?: string
};
*/

export default class Onboarding extends React.PureComponent {
  /*:: mounted: boolean; */
  /*:: props: Props; */
  /*:: state: State; */

  static contextTypes = {
    router: PropTypes.object
  };

  constructor(props /*: Props */) {
    super(props);
    this.state = {
      finished: false,
      skipping: false,
      step: props.organizationsEnabled ? 'organization' : 'token'
    };
  }

  componentDidMount() {
    this.mounted = true;

    // useCapture = true to receive the event before inputs
    window.addEventListener('keydown', this.onKeyDown, true);

    if (!this.props.currentUser.isLoggedIn) {
      handleRequiredAuthentication();
    }
  }

  componentWillUnmount() {
    window.removeEventListener('keydown', this.onKeyDown, true);
    this.mounted = false;
  }

  onKeyDown = (event /*: KeyboardEvent */) => {
    // ESC key
    if (event.keyCode === 27) {
      this.finishOnboarding();
    }
  };

  finishOnboarding = () => {
    this.setState({ skipping: true });
    skipOnboarding().then(
      () => {
        if (this.mounted) {
          this.props.onFinish();

          if (this.state.projectKey) {
            this.context.router.push(getProjectUrl(this.state.projectKey));
          }
        }
      },
      () => {
        if (this.mounted) {
          this.setState({ skipping: false });
        }
      }
    );
  };

  handleTimeout = () => {
    // unset `projectKey` to display a generic "Finish this tutorial" button
    this.setState({ projectKey: undefined });
  };

  handleTokenDone = (token /*: string */) => {
    this.setState({ step: 'analysis', token });
  };

  handleOrganizationDone = (organization /*: string */) => {
    this.setState({ organization, step: 'token' });
  };

  handleTokenOpen = () => this.setState({ step: 'token' });

  handleOrganizationOpen = () => this.setState({ step: 'organization' });

  handleSkipClick = (event /*: Event */) => {
    event.preventDefault();
    this.finishOnboarding();
  };

  handleFinish = (projectKey /*: string | void */) => this.setState({ finished: true, projectKey });

  handleReset = () => this.setState({ finished: false, projectKey: undefined });

  render() {
    if (!this.props.currentUser.isLoggedIn) {
      return null;
    }

    const { organizationsEnabled, sonarCloud } = this.props;
    const { step, token } = this.state;

    let stepNumber = 1;

    const header = translate(sonarCloud ? 'onboarding.header.sonarcloud' : 'onboarding.header');

    return (
      <div className="modal-container">
        <Helmet title={header} titleTemplate="%s" />

        <div className="page page-limited onboarding">
          <header className="page-header">
            <h1 className="page-title">{header}</h1>
            <div className="page-actions">
              {this.state.skipping ? (
                <i className="spinner" />
              ) : (
                <a className="js-skip text-muted" href="#" onClick={this.handleSkipClick}>
                  {translate('tutorials.skip')}
                </a>
              )}
              <p className="note">
                {translate(
                  sonarCloud ? 'tutorials.find_it_back_in_plus' : 'tutorials.find_it_back_in_help'
                )}
              </p>
            </div>
            <div className="page-description">
              {translateWithParameters(
                'onboarding.header.description',
                organizationsEnabled ? 3 : 2
              )}
            </div>
          </header>

          {organizationsEnabled && (
            <OrganizationStep
              currentUser={this.props.currentUser}
              finished={this.state.organization != null}
              onContinue={this.handleOrganizationDone}
              onOpen={this.handleOrganizationOpen}
              open={step === 'organization'}
              stepNumber={stepNumber++}
            />
          )}

          <TokenStep
            currentUser={this.props.currentUser}
            finished={this.state.token != null}
            onContinue={this.handleTokenDone}
            onOpen={this.handleTokenOpen}
            open={step === 'token'}
            stepNumber={stepNumber++}
          />

          <AnalysisStep
            onFinish={this.handleFinish}
            onReset={this.handleReset}
            organization={this.state.organization}
            open={step === 'analysis'}
            sonarCloud={sonarCloud}
            stepNumber={stepNumber}
            token={token}
          />

          {this.state.finished &&
            !this.state.skipping &&
            (this.state.projectKey ? (
              <ProjectWatcher
                onFinish={this.finishOnboarding}
                onTimeout={this.handleTimeout}
                projectKey={this.state.projectKey}
              />
            ) : (
              <footer className="text-right">
                <a className="button" href="#" onClick={this.handleSkipClick}>
                  {translate('tutorials.finish')}
                </a>
              </footer>
            ))}
        </div>
      </div>
    );
  }
}
