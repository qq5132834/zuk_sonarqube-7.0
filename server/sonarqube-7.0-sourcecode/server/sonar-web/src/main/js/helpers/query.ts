/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { isNil, omitBy } from 'lodash';
import { isValidDate, parseDate, toNotSoISOString } from './dates';

export interface RawQuery {
  [x: string]: any;
}

function arraysEqual(a: RawQuery, b: RawQuery): boolean {
  if (a.length !== b.length) {
    return false;
  }
  for (let i = 0; i < a.length; i++) {
    if (a[i] !== b[i]) {
      return false;
    }
  }
  return true;
}

export function queriesEqual(a: RawQuery, b: RawQuery): boolean {
  const keysA = Object.keys(a);
  const keysB = Object.keys(b);

  if (keysA.length !== keysB.length) {
    return false;
  }

  return keysA.every(
    key =>
      Array.isArray(a[key]) && Array.isArray(b[key])
        ? arraysEqual(a[key], b[key])
        : a[key] === b[key]
  );
}

export function cleanQuery(query: RawQuery): RawQuery {
  return omitBy(query, isNil);
}

export function parseAsBoolean(value: string | undefined, defaultValue: boolean = true): boolean {
  return value === 'false' ? false : value === 'true' ? true : defaultValue;
}

export function parseAsDate(value?: string): Date | undefined {
  if (value) {
    const date = parseDate(value);
    if (isValidDate(date)) {
      return date;
    }
  }
  return undefined;
}

export function parseAsFacetMode(facetMode: string): string {
  return facetMode === 'debt' || facetMode === 'effort' ? 'effort' : 'count';
}

export function parseAsString(value: string | undefined): string {
  return value || '';
}

export function parseAsArray(
  value: string | undefined,
  itemParser: (x: string) => string
): string[] {
  return value ? value.split(',').map(itemParser) : [];
}

export function serializeDate(value?: Date): string | undefined {
  if (value != null && value.toISOString) {
    return toNotSoISOString(value);
  }
  return undefined;
}

export function serializeString(value: string): string | undefined {
  return value || undefined;
}

export function serializeStringArray(value: string[] | undefined[]): string | undefined {
  return value && value.length ? value.join() : undefined;
}
