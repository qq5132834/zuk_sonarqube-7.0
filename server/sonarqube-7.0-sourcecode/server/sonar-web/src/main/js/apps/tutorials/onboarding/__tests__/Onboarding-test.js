/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import React from 'react';
import { shallow, mount } from 'enzyme';
import Onboarding from '../Onboarding';
import { click, doAsync } from '../../../../helpers/testUtils';

jest.mock('../../../../api/users', () => ({
  skipOnboarding: () => Promise.resolve()
}));

const currentUser = { login: 'admin', isLoggedIn: true };

it('guides for on-premise', () => {
  const wrapper = shallow(
    <Onboarding
      currentUser={currentUser}
      onFinish={jest.fn()}
      organizationsEnabled={false}
      sonarCloud={false}
    />
  );
  expect(wrapper).toMatchSnapshot();

  // $FlowFixMe
  wrapper.instance().handleTokenDone('abcd1234');
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});

it('guides for sonarcloud', () => {
  const wrapper = shallow(
    <Onboarding
      currentUser={currentUser}
      onFinish={jest.fn()}
      organizationsEnabled={true}
      sonarCloud={true}
    />
  );
  expect(wrapper).toMatchSnapshot();

  // $FlowFixMe
  wrapper.instance().handleOrganizationDone('my-org');
  wrapper.update();
  expect(wrapper).toMatchSnapshot();

  // $FlowFixMe
  wrapper.instance().handleTokenDone('abcd1234');
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});

it('finishes', () => {
  const onFinish = jest.fn();
  const wrapper = mount(
    <Onboarding
      currentUser={currentUser}
      onFinish={onFinish}
      organizationsEnabled={false}
      sonarCloud={false}
    />
  );
  click(wrapper.find('.js-skip'));
  return doAsync(() => {
    expect(onFinish).toBeCalled();
  });
});
