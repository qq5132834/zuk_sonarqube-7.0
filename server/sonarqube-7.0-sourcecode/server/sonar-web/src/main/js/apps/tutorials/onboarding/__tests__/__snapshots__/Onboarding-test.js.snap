// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`guides for on-premise 1`] = `
<div
  className="modal-container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="onboarding.header"
    titleTemplate="%s"
  />
  <div
    className="page page-limited onboarding"
  >
    <header
      className="page-header"
    >
      <h1
        className="page-title"
      >
        onboarding.header
      </h1>
      <div
        className="page-actions"
      >
        <a
          className="js-skip text-muted"
          href="#"
          onClick={[Function]}
        >
          tutorials.skip
        </a>
        <p
          className="note"
        >
          tutorials.find_it_back_in_help
        </p>
      </div>
      <div
        className="page-description"
      >
        onboarding.header.description.2
      </div>
    </header>
    <TokenStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={false}
      onContinue={[Function]}
      onOpen={[Function]}
      open={true}
      stepNumber={1}
    />
    <AnalysisStep
      onFinish={[Function]}
      onReset={[Function]}
      open={false}
      sonarCloud={false}
      stepNumber={2}
    />
  </div>
</div>
`;

exports[`guides for on-premise 2`] = `
<div
  className="modal-container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="onboarding.header"
    titleTemplate="%s"
  />
  <div
    className="page page-limited onboarding"
  >
    <header
      className="page-header"
    >
      <h1
        className="page-title"
      >
        onboarding.header
      </h1>
      <div
        className="page-actions"
      >
        <a
          className="js-skip text-muted"
          href="#"
          onClick={[Function]}
        >
          tutorials.skip
        </a>
        <p
          className="note"
        >
          tutorials.find_it_back_in_help
        </p>
      </div>
      <div
        className="page-description"
      >
        onboarding.header.description.2
      </div>
    </header>
    <TokenStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={true}
      onContinue={[Function]}
      onOpen={[Function]}
      open={false}
      stepNumber={1}
    />
    <AnalysisStep
      onFinish={[Function]}
      onReset={[Function]}
      open={true}
      sonarCloud={false}
      stepNumber={2}
      token="abcd1234"
    />
  </div>
</div>
`;

exports[`guides for sonarcloud 1`] = `
<div
  className="modal-container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="onboarding.header.sonarcloud"
    titleTemplate="%s"
  />
  <div
    className="page page-limited onboarding"
  >
    <header
      className="page-header"
    >
      <h1
        className="page-title"
      >
        onboarding.header.sonarcloud
      </h1>
      <div
        className="page-actions"
      >
        <a
          className="js-skip text-muted"
          href="#"
          onClick={[Function]}
        >
          tutorials.skip
        </a>
        <p
          className="note"
        >
          tutorials.find_it_back_in_plus
        </p>
      </div>
      <div
        className="page-description"
      >
        onboarding.header.description.3
      </div>
    </header>
    <OrganizationStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={false}
      onContinue={[Function]}
      onOpen={[Function]}
      open={true}
      stepNumber={1}
    />
    <TokenStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={false}
      onContinue={[Function]}
      onOpen={[Function]}
      open={false}
      stepNumber={2}
    />
    <AnalysisStep
      onFinish={[Function]}
      onReset={[Function]}
      open={false}
      sonarCloud={true}
      stepNumber={3}
    />
  </div>
</div>
`;

exports[`guides for sonarcloud 2`] = `
<div
  className="modal-container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="onboarding.header.sonarcloud"
    titleTemplate="%s"
  />
  <div
    className="page page-limited onboarding"
  >
    <header
      className="page-header"
    >
      <h1
        className="page-title"
      >
        onboarding.header.sonarcloud
      </h1>
      <div
        className="page-actions"
      >
        <a
          className="js-skip text-muted"
          href="#"
          onClick={[Function]}
        >
          tutorials.skip
        </a>
        <p
          className="note"
        >
          tutorials.find_it_back_in_plus
        </p>
      </div>
      <div
        className="page-description"
      >
        onboarding.header.description.3
      </div>
    </header>
    <OrganizationStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={true}
      onContinue={[Function]}
      onOpen={[Function]}
      open={false}
      stepNumber={1}
    />
    <TokenStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={false}
      onContinue={[Function]}
      onOpen={[Function]}
      open={true}
      stepNumber={2}
    />
    <AnalysisStep
      onFinish={[Function]}
      onReset={[Function]}
      open={false}
      organization="my-org"
      sonarCloud={true}
      stepNumber={3}
    />
  </div>
</div>
`;

exports[`guides for sonarcloud 3`] = `
<div
  className="modal-container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="onboarding.header.sonarcloud"
    titleTemplate="%s"
  />
  <div
    className="page page-limited onboarding"
  >
    <header
      className="page-header"
    >
      <h1
        className="page-title"
      >
        onboarding.header.sonarcloud
      </h1>
      <div
        className="page-actions"
      >
        <a
          className="js-skip text-muted"
          href="#"
          onClick={[Function]}
        >
          tutorials.skip
        </a>
        <p
          className="note"
        >
          tutorials.find_it_back_in_plus
        </p>
      </div>
      <div
        className="page-description"
      >
        onboarding.header.description.3
      </div>
    </header>
    <OrganizationStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={true}
      onContinue={[Function]}
      onOpen={[Function]}
      open={false}
      stepNumber={1}
    />
    <TokenStep
      currentUser={
        Object {
          "isLoggedIn": true,
          "login": "admin",
        }
      }
      finished={true}
      onContinue={[Function]}
      onOpen={[Function]}
      open={false}
      stepNumber={2}
    />
    <AnalysisStep
      onFinish={[Function]}
      onReset={[Function]}
      open={true}
      organization="my-org"
      sonarCloud={true}
      stepNumber={3}
      token="abcd1234"
    />
  </div>
</div>
`;
